desc/vmec_utils.py

```diff
@@ -273,6 +274,83 @@ def ptolemy_linear_transform(desc_modes, vmec_modes=None, helicity=None, NFP=Non
     return matrix, vmec_modes
 
 
+def rfft_to_trig(a, n, axis=-1):
+    """Spectral coefficients of the Nyquist trigonometric interpolant.
+
+    Parameters
+    ----------
+    a : jnp.ndarray
+        Fourier coefficients ``a=rfft(f,norm="forward",axis=axis)``.
+    n : int
+        Spectral resolution of ``a``.
+    axis : int
+        Axis along which coefficients are stored.
+
+    Returns
+    -------
+    h : jnp.ndarray
+        Nyquist trigonometric interpolant coefficients.
+
+        Coefficients are ordered along ``axis`` of size ``n`` to match
+        Vandermonde matrix with order
+        [sin(k𝐱), ..., sin(𝐱), 1, cos(𝐱), ..., cos(k𝐱)].
+        When ``n`` is even the sin(k𝐱) coefficient is zero and is excluded.
+
+    """
+    is_even = (n % 2) == 0
+    # sin(nx) coefficients
+    an = -2 * jnp.flip(
+        take(
+            a.imag,
+            jnp.arange(1, a.shape[axis] - is_even),
```

Comment:
I don't follow this logic completely, so when `is_even` is True, are there any sin coefficients?

Reply:
When `n` is even, instead of returning coefficients that can be dotted with
`[sin(k𝐱), ..., sin(𝐱), 1, cos(𝐱), ..., cos(k𝐱)]`
we return
`[sin([k-1]𝐱), ..., sin(𝐱), 1, cos(𝐱), ..., cos(k𝐱)]`
because the coefficient for `sin(k𝐱)` is always zero in that case. By doing there are always the same number of returned coefficients `n` as there were interpolated parameters `n`.